l1-contracts/src/core/Rollup.sol

```diff
@@ -182,15 +182,16 @@ contract Rollup is Leonidas, IRollup, ITestRollup {
    * @param _signatures - Signatures from the validators
    * @param _body - The body of the L2 block
    */
-  function propose(
+  function proposeWithBody(
```

Comment:
Following some of this work, I think it would be great to get rid of some of the `propose` methods. We do not need all of them. As we will be moving to blobs it seems like we could for the interim go with the "there are always provided bodies" and signatures to get rid of some of the others. 

This is more a comment to myself than to this pr (see #8452).